### PR TITLE
feat: Create self paced and live courses

### LINKED ISSUE #281: Implement 2 types of courses
### Feature Description

1. Self Paced
2. Live Class

### Use Case

_No response_

### Proposed Solution

_No response_

### Alternatives (optional)

_No response_

### Additional Context

_No response_

### Please check the boxes that apply to this feature request.

- [ ] I have searched the existing feature requests to make sure this is not a duplicate.
- [ ] I have provided a detailed description of the requested feature.
- [ ] I have explained the use case or scenario for this feature.
- [ ] I have included any relevant technical details or design suggestions.
- [ ] I understand that this is a suggestion and that there is no guarantee of implementation.

### PR DESCRIPTION
**Pull request by Chifez**
## What does this PR do?
this PR enables organizations and tutors to create self paced or live courses

Fixes #281

https://github.com/rotimi-best/classroomio/assets/69228399/70fe3dae-e131-40fe-8511-0ad1f06f5446

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Run on local host

### Required

- [x] Filled out the "How to test" section in this PR
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the ClassroomIO Docs if changes were necessary

### CODE REVIEW

File: apps/dashboard/src/lib/components/Course/components/Lesson/Exercise/ViewMode.svelte [hunk 1]
```diff
@@ -80,7 +91,89 @@
...
+          let questionAnswer = submissionResponse?.res?.data.find(
+            (answer) => answer.question_id === id
+          );
+
+          console.log('this is the question answer', questionAnswer);
+          if (question_type_id !== QUESTION_TYPE.TEXTAREA) {
+            const answer = $questionnaireMetaData.answers[question.name];
+            let score = points / answer.length;
+            $questionnaireMetaData.grades = {
+              ...$questionnaireMetaData.grades,
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Course/components/Lesson/Exercise/ViewMode.svelte:116]: @Chifez Don't use AI to grade paragraph questions.

I think I mentioned earlier that you should just put paragraph question as a maximum score

File: apps/dashboard/src/lib/components/Course/components/Navigation/index.svelte [hunk 2]
```diff
@@ -17,6 +17,7 @@
   import { getIsLessonComplete } from '../Lesson/functions';
   import { isFreePlan } from '$lib/utils/store/org';
   import { t } from '$lib/utils/functions/translations';
+  import { COURSE_TYPE_ENUM } from '$lib/components/Courses/constants';
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Course/components/Navigation/index.svelte]: Please rename `COURSE_TYPE_ENUM` to `COURSE_TYPE`

File: apps/dashboard/src/lib/components/CourseContainer/index.svelte [hunk 3]
```diff
@@ -29,6 +29,7 @@
     const { data: _data } = await fetchCourse(courseId);
 
     if (_data) {
+      $course.course_type = _data.course_type;
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/CourseContainer/index.svelte]: Why is this here?

Why isn't the logic in `setCourse` below?

File: apps/dashboard/src/lib/components/Courses/components/NewCourseModal/index.svelte [hunk 4]
```diff
@@ -13,20 +14,35 @@
...
   import { t } from '$lib/utils/functions/translations';
+  import { COURSE_TYPE_ENUM } from '../../constants';
 
   let isLoading = false;
   let errors = {
     title: '',
     description: ''
   };
 
+  const COURSE_TYPE = [
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Courses/components/NewCourseModal/index.svelte]: This should be `COURSE_TYPES` because it is an array

File: apps/dashboard/src/lib/components/Courses/components/NewCourseModal/index.svelte [hunk 5]
```diff
@@ -13,20 +14,35 @@
...
+    {
+      id: COURSE_TYPE_ENUM.LIVE_CLASS,
+      text: 'live class'
+    },
+    {
+      id: COURSE_TYPE_ENUM.SELF_PACED,
+      text: 'self paced'
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Courses/components/NewCourseModal/index.svelte]: Shouldn't the `text` be separated by an underscore? like `live_class`

**Review by rotimi-best — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/lib/components/Courses/components/NewCourseModal/index.svelte (hunk 5 above)
Comment (Chifez) [reply apps/dashboard/src/lib/components/Courses/components/NewCourseModal/index.svelte]: no because,that is the text that gets displayed in the UI

File: apps/dashboard/src/lib/components/Courses/components/NewCourseModal/index.svelte (hunk 4 above)
Comment (Chifez) [reply apps/dashboard/src/lib/components/Courses/components/NewCourseModal/index.svelte]: alright

File: apps/dashboard/src/lib/components/CourseContainer/index.svelte (hunk 3 above)
Comment (Chifez) [reply apps/dashboard/src/lib/components/CourseContainer/index.svelte]: i didn't even think of that initially, but looking at it now that logic isn't needed because the setcourse function already sets the local course store to the data being passed in

File: apps/dashboard/src/lib/components/Course/components/Lesson/Exercise/ViewMode.svelte (hunk 1 above)
Comment (Chifez) [reply apps/dashboard/src/lib/components/Course/components/Lesson/Exercise/ViewMode.svelte:116]: yea you did but i suggested that we use the AI logic for auto grading and you agreed...i don't think it is right to give out the maximum score for paragraph question regardless of the answer provided, it only gives a false positive to student taking a self paced course

File: apps/dashboard/src/lib/components/Courses/components/NewCourseModal/index.svelte (hunk 5 above)
Comment (Chifez) [reply apps/dashboard/src/lib/components/Courses/components/NewCourseModal/index.svelte]: more importantly i might have forgotten to put the translations here

File: apps/dashboard/src/lib/components/Course/components/Lesson/Exercise/ViewMode.svelte (hunk 1 above)
Comment (rotimi-best) [reply apps/dashboard/src/lib/components/Course/components/Lesson/Exercise/ViewMode.svelte:116]: Yeah I get that but it can add more overhead on our AI functionality if we have to grade all paragraph questions from a cost perspective

Besides self paced courses are for students to take the course how they see fit. No one is verifying if they got the right answer or not. 

In addition there should be more radio or checkbox questions because the teacher isn't grading.

My point here is that the score is less important for self paced courses cause the idea is that the creator doesn't want to monitor the student

Comment (rotimi-best) [reply apps/dashboard/src/lib/components/Course/components/Lesson/Exercise/ViewMode.svelte:116]: @Chifez Please let's do this first on Monday
